std/mem.zig

```diff
@@ -412,6 +412,9 @@ test "mem.indexOf" {
 /// the result.
 /// See also ::readIntBE or ::readIntLE.
 pub fn readInt(bytes: []const u8, comptime T: type, endian: builtin.Endian) T {
+    if (comptime T.bit_count < 8) {
+      return @compileError("Only integer types with bitwidth >= 8 are allowed; passed " ++ @typeName(T));
```

Comment:
I think the API of this function is confusing and should potentially be changed, but as is, this safety logic can't happen at compile time. T is just the return type; `bytes.len` is the number of bytes in the integer. So the safety violation happens if `T.bit_count < bytes.len * 8`, which is a runtime value. That can just be asserted.

Reply:
Hey Andrew, did you see my comment above? if `T.bit_count < 8` we run into the following error:

```
/Users/cfkk/Source/zig2/std/mem.zig:478:30: error: RHS of shift is too large for LHS type
            result = (result << 8) | T(bytes[i]);
                             ^
/Users/cfkk/Source/zig2/std/mem.zig:506:21: note: called from here
    assert(readIntBE(u7, []u8{0x32}) == 0x32);
                    ^
/Users/cfkk/Source/zig2/std/mem.zig:478:33: note: value 8 cannot fit into type u3
            result = (result << 8) | T(bytes[i]);
                                ^
/Users/cfkk/Source/zig2/std/mem.zig:478:39: error: expected type 'u7', found 'u8'
            result = (result << 8) | T(bytes[i]);
                                      ^
/Users/cfkk/Source/zig2/std/mem.zig:506:21: note: called from here
    assert(readIntBE(u7, []u8{0x32}) == 0x32);
```